openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -497,27 +497,37 @@ def test_create_taxonomy(self, user_attr: str, expected_status: int) -> None:
             if user_attr == "staffA":
                 assert response.data["orgs"] == [self.orgA.short_name]
 
-    def test_list_taxonomy_query_count(self):
+    @ddt.data(
+        ('staff', 11),
+        ("content_creatorA", 16),
+        ("library_staffA", 16),
+        ("library_userA", 16),
+        ("instructorA", 16),
+        ("course_instructorA", 16),
+        ("course_staffA", 16),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_query_count(self, user_attr: str, expected_queries: int):
         """
         Test how many queries are used when retrieving taxonomies and permissions
         """
-        url = TAXONOMY_ORG_LIST_URL + f'?org=${self.orgA.short_name}&enabled=true'
-
-        self.client.force_authenticate(user=self.staff)
-        with self.assertNumQueries(16):  # TODO Why so many queries?
+        url = TAXONOMY_ORG_LIST_URL + f'?org={self.orgA.short_name}&enabled=true'
+        user = getattr(self, user_attr)
+        self.client.force_authenticate(user=user)
+        with self.assertNumQueries(expected_queries):
```

Comment:
So the number of queries here didn't change from 16 in most cases? It that number fixed or does it increase with each additional taxonomy in the result list?

Reply:
Yes unfortunately, 16 queries was as low as I could get them.
But that number is constant -- it used to be relative to the number of TaxonomyOrgs on each returned Taxonomy.

Reply:
(Expand the notes in the PR description for details about what queries are still being executed.)

Reply:
Hi @bradenmacdonald , did you find anything else that I need to fix in this PR?